erts/emulator/beam/external.c

```diff
@@ -3142,6 +3224,87 @@ enc_term_int(TTBEncodeContext* ctx, ErtsAtomCacheMap *acmp, Eterm obj, byte* ep,
 		obj = CAR(ptr);
 	    }
 	    break;
+	case ENC_STORE_MAP_ELEMENT:  /* option `deterministic` */
+	    if (is_list(obj)) { /* leaf node [K|V] */
+		ptr = list_val(obj);
+                *next_map_element++ = CAR(ptr);
+                *next_map_element++ = CDR(ptr);
+                goto outer_loop;
+	    }
+	    break;
+	case ENC_START_SORTING_MAP: /* option `deterministic` */
+            {
+                long num_reductions = r;
+
+                n = next_map_element - map_array;
+                ASSERT(n > MAP_SMALL_MAP_LIMIT);
+                if (ctx == NULL) {
+                    /* No context means that the external representation of term
+                     * being encoded will fit in a heap binary (64 bytes). This
+                     * can only happen in the DEBUG build of the runtime system
+                     * where maps with more than 3 elements are large maps. */
+                    ASSERT(n < 64); /* Conservative assertion. */
+                    qsort(map_array, n/2, 2*sizeof(Eterm),
+                          (int (*)(const void *, const void *)) map_key_compare);
+                    WSTACK_PUSH2(s, ENC_PUSH_SORTED_MAP, THE_NON_VALUE);
+                    goto outer_loop;
+                } else {
+                    /* Use yieldable qsort since the number of elements
+                     * in the map could be huge. */
+                    num_reductions = r;
+                    ctx->ycf_yield_state = NULL;
+                    yielding_qsort_ycf_gen_yielding(&num_reductions,
+                                                    &ctx->ycf_yield_state,
+                                                    NULL,
+                                                    ycf_yield_alloc,
+                                                    ycf_yield_free,
+                                                    NULL,
+                                                    0,
+                                                    NULL,
+                                                    map_array, n);
```

Comment:
I think there is a potential for a memory leak as I don't think `yielding_qsort_ycf_gen_destroy` is called anywhere. A memory leak can happen if the process calling term_to_binary/2 is killed when yielded in the `yielding_qsort_ycf_gen_yielding` call. Then `yielding_qsort_ycf_gen_yielding` has allocated memory that will never be freed.